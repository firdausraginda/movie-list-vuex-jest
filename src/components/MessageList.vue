<template>
  <div class="message-list">
    <ul>
      <li
        v-for="message in messagesProp"
        :key="message"
        class="message-list-ul-li"
      >
        {{ message }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  data() {
    return {};
  },
  props: {
    messagesProp: {
      type: Array,
      default: null
    }
  }
};
</script>

<style lang="scss" scoped>
.message-list {
  .message-list-ul-li {
    color: #42b983;
  }
}
</style>