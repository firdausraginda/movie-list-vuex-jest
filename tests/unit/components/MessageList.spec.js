import { shallowMount } from '@vue/test-utils'
import MessageList from '../../../src/components/MessageList.vue'

describe('MessageList.spec.js', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(MessageList, {
            propsData: {
                messagesProp: ['Cat', 'Monkey']
            }
        })
    })
    
    test('is MessageList component', () => {
        expect(wrapper.is(MessageList)).toBe(true)
        expect(wrapper.is('.message-list')).toBe(true)
    })
    
    test('message property has array type of data', () => {
        expect(typeof wrapper.vm.messagesProp).toBe('object')
    })

    test('has received ["Cat", "Monkey"] as the messagesProp property', () => {
        expect(wrapper.props().messagesProp).toEqual(["Cat", "Monkey"])
    })

    test('has the expected html structure', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

})