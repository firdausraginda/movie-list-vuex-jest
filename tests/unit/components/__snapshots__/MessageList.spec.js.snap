// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.spec.js has the expected html structure 1`] = `
<div
  class="message-list"
>
  <ul>
    <li
      class="message-list-ul-li"
    >
      
      Cat
    
    </li>
    <li
      class="message-list-ul-li"
    >
      
      Monkey
    
    </li>
  </ul>
</div>
`;
